MAINT: No need to check for check for FPEs in casts to/from object

seberg opened the pull request:
Since these go via Python (in some form) and Python doesn't use FPEs we can be sure that we don't need to check for FPEs.

Note that while it hides *almost always* spurious FPEs seen on some platforms, there could be certain chains or multiple cast situations where FPEs are checked for other reasons and the spurious FPE will show up.

So it "somewhat":

Closes gh-28351

seberg committed:
MAINT: No need to check for check for FPEs in casts to/from object

Since these go via Python (in some form) and Python doesn't use FPEs
we can be sure that we don't need to check for FPEs.

Note that while it hides *almost always* spurious FPEs seen on some
platforms, there could be certain chains or multiple cast situations
where FPEs are checked for other reasons and the spurious FPE will
show up.

So it "somewhat":

Closes gh-28351

ngoldbaum commented:
I’ll try to see if this fixes the test on my FreeBSD VM tomorrow since it looks like we’re out of compute credits.

seberg commented:
Ping @rgommers I think we have been out of credits for a little bit, could you recharge it some time?

rgommers commented:
> Ping @rgommers I think we have been out of credits for a little bit, could you recharge it some time?

Done, sorry about that. Added 200 credits, that should last us 4-5 months at the current rate.

seberg committed:
MAINT: Follow-up, got the wrong place (the other is OK).

seberg commented:
@rgommers thanks!

@ngoldbaum FWIW, I got the wrong place there (both places is fine) and CI is running now, so no need to check locally.

ngoldbaum commented:
I was actually in the middle of building NumPy in my VM to understand why the tests failed :)

Thanks for following up. Assuming this fixes the tests let's close the other PR skipping the test on freebsd and merge this.

mhvk reviewed (commented):
I'd tend to add comments about why the flags are added, since the PR text itself makes obvious it is not obvious

EDIT: Just to be sure: I meant add comments to the changed lines.

seberg committed:
DOC: Add a small comment as per review request

I don't think it needs a comment that we can do this, but maybe it
is nice to say that there was a reason for it.

charris commented:
Thanks Sebastian.
